magento/magento2
# magento/magento2#9720 Menu item dependencies (dependsOnModule, depend…

### Description
dependsOnModule, dependsOnConfig, toolTip should match casing mandated by menu.xsd
please also [note](https://github.com/magento/magento2/blob/2.2.0-RC2.1/app/code/Magento/Backend/Test/Unit/Model/_files/menu_merged.php) which already has the correct values.


### Fixed issues
magento/magento2#9720: Menu item dependencies 
### Manual testing scenarios
1. setup a module with 

```
    
    <?xml version="1.0"?>
    <config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Backend:etc/menu.xsd">
        <menu>
            <add id="Fooman_Example::test"
                 parent="Magento_Sales::sales"
                 resource="Fooman_Example::test"
                 module="Fooman_Example"
                 title="Fooman Example"
                 sortOrder="20"
                 action="foomanexample/test"
                 dependsOnConfig="foomanexample/menu/display_test"/>
        </menu>
    </config>
```

2. Make sure foomanexample/menu/display_test is a false value
3. Observe that the menu item is not visible

Backport of https://github.com/magento/magento2/pull/10543

### Contribution checklist
